plugins/formal-verification/src/org/jetbrains/kotlin/formver/embeddings/ExpEmbedding.kt

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.embeddings
+
+import org.jetbrains.kotlin.formver.domains.NullableDomain
+import org.jetbrains.kotlin.formver.domains.UnitDomain
+import org.jetbrains.kotlin.formver.scala.silicon.ast.*
+import org.jetbrains.kotlin.formver.scala.toScalaBigInt
+import viper.silver.ast.DomainType
+import javax.sound.sampled.LineEvent.Type
+
+sealed interface ExpEmbedding {
+    val type: TypeEmbedding
+    val viperExp: Exp
```

Comment:
I'm not a fan of this terminology because it isn't clear whether it goes to stage 3 or stage 4...
Perhaps we should rename `IntoViper` into `IntoSilver`, since Silver is the actual Scala implementation of the Viper Intermediate Language?  That would let us call stage 3 "Viper", since it's our representation of VIR...